### ISSUE TITLE
Nom français

### ISSUE THREAD

**Issue by jbdesbas** [PnX-SI/GeoNature#699]
Salut, 
J'ai remarqué que les nom français indiqué dans Taxhub (table bib_noms) ne sont pas utilisés dans la synthèse (en recherche + fiche) et le module de validation.
Par contre ils sont bien utilisés dans OccTax.
Y a t'il une config qui m'aurait échappé quelque part ? Les noms français du Taxref sont parfois tirés par les cheveux..

De plus, je n'ai pas trouvé d'API dans taxhub permettant d'obtenir le nom français à partir du cd_nom (seulement à partir de l'id_nom)

**Comment by TheoLechemia** [comment]
Non, en effet Validation et Synthese s’appuie directement sur Taxref pour le nom_français. Voir la table "taxons_synthese_autocomplete" qui liste tous les taxons présents dans la synthese. Cette table est maintenu en trigger par les observations saisies dans la synthese. 
En fait on veut supprimer à moyen terme l'utilisation de bib_noms, du coup les dernières choses développés ne s’appuie plus dessus (Occtax est un peu plus vieux, et dépend d'un route de Taxhub directement pour la liste des taxons). 
Mais on retient l'idée de pouvoir customiser le nom français dans la futur suppression de bib_noms.

**Comment by jbdesbas** [comment]
OK je comprend mieux merci. Garder la possibilité d'écraser le nom français taxref par un autre me parait essentiel aussi.
Tu as déjà une idée de la forme que ca prendra ?

**Comment by TheoLechemia** [comment]
Non on n'y a pas trop réfléchi encore

**Comment by jbdesbas** [comment]
@TheoLechemia un petit [trick](https://gist.github.com/jbdesbas/871804ab52a47dcf1093343aa3a2683b) que j'utilise dans ma base pour modifier les noms français.
La modification se fait via une table _cd_nom / cd_ref / nom_vern_ . Un trigger repporte la modification dans la table _taxref_ et _autocomplete_.
J'ai choisi de ne rendre modifiable que le nom français et le cd_ref car se sont pour moi les principaux points de désaccord que les experts locaux peuvent avoir avec le taxref.

__Avantages__ : 
On peut personnaliser les nom_vern et les cd_ref 
On stock les points de divergences avec les taxref (pour pouvoir les ré-intégrer après une MAJ du taxref)

__Inconvénients__ : 
On écrase directement les données du taxref (mais est-ce un gros problème, car les cd_nom et nom latins sont conservés).

Je ne sais pas s'il y aurait un intérêt à intégrer ça, ou si c'est trop "bricolage"

**Comment by camillemonchicourt** [comment]
On avait justement créée la table bib_noms notamment pour pouvoir surcoucher les noms français sans toucher à la table Taxref.
Je pense qu'il faut éviter.
A voir quel mécanisme de surcouche on peut trouver si on supprime bib_noms

**Comment by jbdesbas** [comment]
Effectivement j’étais parti du principe que _bib_noms_ allait disparaître. Pour l'instant j'utilise ça faute de mieux mais ca n'est pas idéal.
Je pense que ca peut être sympa de prévoir un mécanisme d'ajout d'espèces hors taxref en plus de la surcouche.
